multimodal/src/autogluon/multimodal/models/huggingface_text.py

```diff
@@ -191,25 +202,42 @@ def forward(
             raise NotImplementedError(f"Pooling mode={self.pooling_mode} is not supported.")
 
         logits = self.head(pooled_features)
-
-        ret = {COLUMN_FEATURES: {FEATURES: {}, MASKS: {}}}
+        last_hidden_state = outputs.last_hidden_state
+
+        batch = {
+            self.text_token_ids_key: text_token_ids,
+            self.text_segment_ids_key: text_segment_ids,
+            self.text_valid_length_key: text_valid_length,
+        }
+        if text_column_names:
+            assert len(text_column_names) == len(text_column_indices), "invalid text column inputs"
+            batch.update(**dict(zip(text_column_names, text_column_indices)))
         column_features, column_feature_masks = get_column_features(
             batch=batch,
             column_name_prefix=self.text_column_prefix,
-            features=outputs.last_hidden_state,
+            features=last_hidden_state,
             valid_lengths=text_valid_length,
             cls_feature=pooled_features,
         )
-        ret[COLUMN_FEATURES][FEATURES].update(column_features)
-        ret[COLUMN_FEATURES][MASKS].update(column_feature_masks)
-
-        ret.update(
-            {
-                LOGITS: logits,
-                FEATURES: pooled_features,
-            }
-        )
 
+        if column_features == {} or column_feature_masks == {}:
+            return pooled_features, logits
+        else:
+            return pooled_features, logits, column_features, column_feature_masks
```

Comment:
Other models (`mmdet_image`, `t_few`) still use dictionary as input/output. Shall we force the consistency here? We could either make changes on a new function `forward_onnx` instead, or change the input/output of all other models as well.